apps/rule-manager/app/model/PaginatedResponse.scala

```diff
@@ -0,0 +1,19 @@
+package model
+
+import play.api.libs.json.{Format, Json}
+
+import scala.annotation.nowarn
+
+case class PaginatedResponse[Data](
+    data: List[Data],
+    pageSize: Int,
+    page: Int,
+    pages: Int,
+    total: Int
+)
+
+object PaginatedResponse {
+  @nowarn
+  implicit def format[Data](implicit _underlying: Format[Data]): Format[PaginatedResponse[Data]] =
```

Comment:
grid does this with the slightly terser 
```suggestion
  implicit def format[Data: Format]: Format[PaginatedResponse[Data]] =
```
( I hadn't internalised exactly how this syntax works before - TIL this is a [_context bound_](https://docs.scala-lang.org/scala3/book/ca-context-bounds.html) which Scala compiler will convert into essentially the same as your version - but without the "unused implicit" warning :) )

Reply:
So much nicer – thank you!